packages/@react-aria/focus/stories/FocusScope.stories.tsx

```diff
@@ -46,35 +47,47 @@ function MaybePortal({children, usePortal}: { children: ReactNode, usePortal: bo
   );
 }
 
-function NestedDialog({onClose, usePortal}: {onClose: VoidFunction, usePortal: boolean}) {
+function NestedDialog({onClose, usePortal, contain}: {onClose: VoidFunction, usePortal: boolean, contain: boolean}) {
   let [open, setOpen] = useState(false);
+  let [showNew, setShowNew] = useState(false);
+  let onKeyDown = (e) => {
+    if (e.key === 'Escape') {
+      e.stopPropagation();
+      onClose();
+    }
+  };
 
   return (
     <MaybePortal usePortal={usePortal}>
-      <FocusScope contain restoreFocus autoFocus>
-        <div>
-          <input />
-
-          <input />
-
-          <input />
-
-          <button type="button" onClick={() => setOpen(true)}>
-            Open dialog
-          </button>
-
-          <button type="button" onClick={onClose}>
-            close
-          </button>
-
-          {open && <NestedDialog onClose={() => setOpen(false)} usePortal={usePortal} />}
-        </div>
+      <FocusScope contain={contain} restoreFocus autoFocus>
+        {!showNew && (
+          <div role="dialog" onKeyDown={onKeyDown}>
+            <input />
+            <input />
+            <input />
+            <button onClick={() => setShowNew(true)}>replace focusscope children</button>
+            <button type="button" onClick={() => setOpen(true)}>
+              Open dialog
+            </button>
+            <button type="button" onClick={onClose}>
+              close
+            </button>
+            {open && <NestedDialog contain={contain} onClose={() => setOpen(false)} usePortal={usePortal} />}
+          </div>
+        )}
+        {showNew && (
+          <div role="dialog" onKeyDown={onKeyDown}>
+            <input />
+            <input autoFocus />
```

Comment:
`autoFocus` on this `input` means that focus will be maintained within the FocusScope when the children change, however, without the `autoFocus` prop, focus will be lost to the `document.body`. https://github.com/adobe/react-spectrum/pull/2763 has been opened to account for this and keep focus within the FocusScope when `contain` is `true`.

Reply:
yep, #2763 is the reason we aren't addressing this issue here